### PR TITLE
fixes #20407 - adds two new options

### PR DESCRIPTION
**Pull request by mriedel-pgdx**
Fixes #20407 - adds two new options.
Adds ability to configure ansible_become & ansible_ssh_private_key_file options to foreman_ansible plugin. IDE also edited whitespace in ansible.rb file.

### CODE REVIEW

File: app/models/setting/ansible.rb [hunk 1]
```diff
@@ -37,6 +46,15 @@ def load_defaults
               N_('Password')
             ),
             set(
+              'ansible_ssh_private_key_file',
+              N_('Use this to supply a path to an SSH Private Key '\
+                 'that Ansible will use in lieu of a password '\
+                 'Override with "ansible_ssh_private_key_file"'\
```
Comment (xprazak2) [inline app/models/setting/ansible.rb]: nit: needs a space after the closing `"`

File: app/models/setting/ansible.rb [hunk 2]
```diff
@@ -15,20 +15,29 @@ def load_defaults
...
+                 '"ansible_user"'),
               'root',
               N_('User')
             ),
             set(
+              'ansible_become',
+              N_('Foreman will use the sudo command to run roles on hosts '\
+                 'You can override this on hosts by adding a parameter '\
+                 '"ansible_become"'),
+              'True',
```
Comment (xprazak2) [inline app/models/setting/ansible.rb]: try replacing `'True'` with `true`, our helper should recognize the boolean and use dropdropdown instead of text field

File: app/services/foreman_ansible/inventory_creator.rb [hunk 3]
```diff
@@ -38,10 +38,18 @@ def connection_params(host)
...
         'ansible_user' => host_user(host),
-        'ansible_ssh_pass' => host_ssh_pass(host),
+        'ansible_become' => host_become(host),
         'ansible_connection' => connection_type(host),
         'ansible_winrm_server_cert_validation' => winrm_cert_validation(host)
       }
+      # Check if private_key is defined, if it is use that. Otherwise
+      # use ssh_pass
+      if !host.host_params['ansible_ssh_private_key_file'].empty? ||
+          !Setting[:private_key].empty?
```
Comment (xprazak2) [inline app/services/foreman_ansible/inventory_creator.rb]: I think this should be `Setting[:ansible_ssh_private_key_file]`. Note that condition will be true out of the box since you set the default value to `/tmp/private_key_file`, so I would recommend keeping it blank. You can also take advantage of your `host_private_key_file` method and change it to:

```ruby
unless host_private_key_file(host).empty?
# ...
else
# ...
end
```

**Comment by xprazak2** [comment]
[test]

**Comment by xprazak2** [comment]
Thank you for your patch, I left a few comments inline, but it looks very nice!

**Comment by mriedel-pgdx** [comment]
Hi @xprazak2 ! Thanks for the feedback, and catching the 'private_key' I missed!

I've made the changes you suggested, except the "unless" because I'm using an "else" with it. My IDE raised a warning on that, and evidently there's some [controversy](http://www.railstips.org/blog/archives/2008/12/01/unless-the-abused-ruby-conditional/) over doing things that way.

Also, maybe this is a Foreman bug, but in my testing empty strings ('') as defaults caused the setting to disappear from the Ansible settings page in Foreman. The existing `Default verbosity level` setting does not appear in Foreman 1.15.2, and the same thing happened with `ansible_ssh_private_key_file`. Regardless, I've updated that too.

File: app/services/foreman_ansible/inventory_creator.rb [hunk 4]
```diff
@@ -38,10 +38,18 @@ def connection_params(host)
...
         'ansible_user' => host_user(host),
-        'ansible_ssh_pass' => host_ssh_pass(host),
+        'ansible_become' => host_become(host),
         'ansible_connection' => connection_type(host),
         'ansible_winrm_server_cert_validation' => winrm_cert_validation(host)
       }
+      # Check if private_key is defined, if it is use that. Otherwise
+      # use ssh_pass
+      if !host.host_params['ansible_ssh_private_key_file'].empty? ||
+          Setting[:ansible_ssh_private_key_file].empty?
```
Comment (xprazak2) [inline app/services/foreman_ansible/inventory_creator.rb]: I think the `!` got lost, it was here before. You can keep `if`, I do not mind either way. My main point was that you can re-use the logic from `host_private_key_file` method and be more DRY:

```ruby
if !host_private_key_file(host).empty?
```

**Comment by xprazak2** [comment]
I solved the missing setting mystery. We have to add a setting with a blank default into whitelist. If you add:

```ruby
Setting::BLANK_ATTRS.push('ansible_ssh_private_key_file')
```

the setting will turn up. Here is an [example](https://github.com/theforeman/foreman/blob/develop/app/models/setting/provisioning.rb#L15) how we use it in other places.

**Comment by mriedel-pgdx** [comment]
@dLobatog Thanks! I merged the changes in and recommitted.

**Comment by dLobatog** [comment]
Thanks @mriedel-pgdx! Note you can use any arbitrary ansible CLI option using the latest Foreman Ansible version (only available in git now, I will release a package next week probably). 

Basically any `ansible_` param will be passed on to the playbook so it can be used as a connection option, similar to what you did here, except other parameters won't have a global setting.
